core/src/main/java/com/amplifyframework/core/model/scalar/AWSTime.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.core.model.scalar;
+
+import androidx.annotation.NonNull;
+import androidx.core.util.ObjectsCompat;
+
+import org.threeten.bp.DateTimeUtils;
+import org.threeten.bp.Instant;
+import org.threeten.bp.LocalDate;
+import org.threeten.bp.LocalTime;
+import org.threeten.bp.OffsetDateTime;
+import org.threeten.bp.OffsetTime;
+import org.threeten.bp.ZoneOffset;
+import org.threeten.bp.format.DateTimeFormatter;
+import org.threeten.bp.format.DateTimeParseException;
+
+import java.util.Date;
+
+/**
+ * Represents a valid extended ISO-8601 Time string, with an optional timezone offset.
+ * <p>
+ * hh:mm:ss.sss±hh:mm:ss
+ * OR
+ * hh:mm:ss.sss
+ * <p>
+ * https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html#appsync-defined-scalars
+ */
+public final class AWSTime {
+    private final LocalTime localTime;
+    private final ZoneOffset zoneOffset;
+
+    public AWSTime(@NonNull Date date) {
+        this.zoneOffset = null;
+        this.localTime = Instant.ofEpochMilli(date.getTime()).atOffset(ZoneOffset.UTC).toLocalTime();
+    }
+
+    public AWSTime(@NonNull Date date, int offsetInSeconds) {
+        this.zoneOffset = ZoneOffset.ofTotalSeconds(offsetInSeconds);
+        this.localTime = Instant.ofEpochMilli(date.getTime()).atOffset(this.zoneOffset).toLocalTime();
+    }
+
+    public AWSTime(@NonNull String text) {
+        LocalTime lt;
+        ZoneOffset zo;
```

Comment:
The current checkstyle will flag variable names under lenght 3, for public values and method arguments. It does _not_ check implementation identifiers like this one. I'll leave it up to you, but `localTime` and `zoneOffset` would be more consistent with the style in the rest of the current code base.

Reply:
I don't prefer one style versus the other, but for consistency with current code base style, I'll make this change!